### PR TITLE
feat: added support for sending custom context

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

> added support for sending custom context using RudderOption object's putCustomContext()

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [x] Changes have been reviewed by at least one other engineer
- [x] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: packages/plugins/rudder_plugin_interface/lib/src/models/rudder_option.dart [hunk 1]
```diff
@@ -3,17 +3,17 @@ import 'rudder_integration.dart';
 
 // we left fetching the external ids from the scratch here
 class RudderOption {
-  List<Map<String, dynamic>>? externalIds;
-  Map<String, Object>? integrations;
+  List<Map<String, String>>? externalIds;
+  Map<String, bool>? integrations;
```
Comment (bardisg) [inline packages/plugins/rudder_plugin_interface/lib/src/models/rudder_option.dart:7]: is thsi boolean or we can pass boolean and object for configuration of integration like in JS SDK?

Comment (desusai7) [reply packages/plugins/rudder_plugin_interface/lib/src/models/rudder_option.dart:7]: In Mobile, we do have the support for boolean only

**Comment by desusai7** [comment]
- Added Support for sending custom context to the web platform as well
- Added Support for sending an object value for enabling an Integration for the web platform
- Added Unit test cases for checking if the custom context and integrations object are properly prepared for both web and mobile platforms

File: packages/plugins/rudder_plugin_interface/lib/src/models/rudder_option.dart (hunk 1 above)
Comment (desusai7) [reply packages/plugins/rudder_plugin_interface/lib/src/models/rudder_option.dart]: Now added support for sending object for configuration of integration like in JS SDK

**Review by bardisg — APPROVED** [review]

**Review by bardisg — APPROVED** [review]

**Review by pallabmaiti — APPROVED** [review]

**Pull request merged**